inputters/markdown_spec.lua

```diff
@@ -0,0 +1,35 @@
+SILE = require("core.sile")
+SILE.backend = "dummy"
+SILE.init()
+SILE.utilities.error = error
+
+describe("#Markdown #inputter", function ()
+  local inputter = SILE.inputters.markdown()
+
+  describe("should parse", function ()
+
+    it("headers", function()
+      local t = inputter:parse("# foo bar\n")[1][1][1][1]
+      assert.is.equal("chapter", t.command)
```

Comment:
I have hard times too here. Now the parser doesn't even go for "chapter" here, because the book chapter we have doesn't have thee necessary support (such as linking to the "id" / hash) - So I looked at your pandoc package too, acknowledge the fact, and parsed to "markdown:internal:header" (at this point...), leveraging around the book "chapter" and even providing a fallback (so the package/inputter works even with the "plain" class, or at least degrades gracefully).

That is to say, I am unsure we can test the parser _that_ way. I have some idea how it could be achieved, but these are just ideas at this stage (i.e. if we don't eventually go for end-to-end tests). Sometimes the nesting changes too (so the `[1][1][1][1]` just above would fail, albeit the end-result would be the same). One of the idea is to compare generated AST from a "simpler" version (alike to what debutAst or autodoc do = rebuild a human-readable version of the AST and just compare), but it is not that straight-forward.

Reply:
These tests shouldn't be considered final, I was just spinning up the testing framework so we had someplace to start from. I had to make sure the inputter was doing *something*. I wasn't satisfied with the extra nesting etc. either, but it's just testing to what is in the AST. If we decide what the parser is putting in the AST needs to change, this test should change.

Reply:
> alike to what debutAst or autodoc do = rebuild a human-readable version of the AST and just compare)

One idea I had here, for instance, is a SILE option that would just dump a human-readable AST and stop there (without going to outputters, etc.). This "human-readable'" form would not be that of the current debugAST, but closer to what autodoc does (er... without the color stuff etc. obviously ^^), so we'd get a SIL-representation of the AST (so probably _less_ readable than the original input if it was in SIL-language, e.g. without line breaks, useless spaces etc.),  that we could even reprocess in debug and compare for identity.

Reply:
BTW:
> My next priority is to get the typesetter object untangled
That would be cool. But give it a bit of time and thinking too... I am currently using a custom typesetter (not committed to my repo, because it breaks too many things... It's a testing branch only ^^), and I start to see where it goes wrong:
- We are absolutizing units too early or too late ^^ (e.g. a kern is immediately "absolutized"
- the "pushback" does weird things with lrskips... (but my own reimplementation breaks elsewhere, so is not really better... or possibly worse ;)

Reply:
Closing this remark. I'm adding tests based on a "reconstructed SIL-like" textual representation from the AST, so expectations are easier to check.